app/controllers/teammates_controller.rb

```diff
@@ -0,0 +1,58 @@
+class TeammatesController < ApplicationController
+  before_action :require_invitation
+  before_action :require_pending
+  rescue_from ActiveRecord::RecordNotFound, :with => :incorrect_token
+
+  def accept
+    if !current_user
```

Comment:
It's generally preferred to do unless vs if !.  The other thing you can do is flip your if and else block so that you run the current_user branch of the if/else and have the else be the no current_user path.